presto-mysql/src/test/java/io/prestosql/plugin/mysql/MySqlQueryRunner.java

```diff
@@ -46,42 +45,38 @@ public static QueryRunner createMySqlQueryRunner(TestingMySqlServer server, Map<
             throws Exception
     {
         try {
-            return createMySqlQueryRunner(server.getJdbcUrl(), connectorProperties, tables);
-        }
-        catch (Throwable e) {
-            closeAllSuppress(e, server);
-            throw e;
-        }
-    }
+            DistributedQueryRunner queryRunner = null;
+            try {
+                queryRunner = new DistributedQueryRunner(createSession(), 3);
 
-    public static QueryRunner createMySqlQueryRunner(String jdbcUrl, Map<String, String> connectorProperties, Iterable<TpchTable<?>> tables)
-            throws Exception
-    {
-        DistributedQueryRunner queryRunner = null;
-        try {
-            queryRunner = new DistributedQueryRunner(createSession(), 3);
+                queryRunner.installPlugin(new TpchPlugin());
+                queryRunner.createCatalog("tpch", "tpch");
 
-            queryRunner.installPlugin(new TpchPlugin());
-            queryRunner.createCatalog("tpch", "tpch");
+                connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
+                connectorProperties.putIfAbsent("connection-url", server.getJdbcUrl());
+                connectorProperties.putIfAbsent("allow-drop-table", "true");
+                connectorProperties.putIfAbsent("connection-user", server.getUser());
+                connectorProperties.putIfAbsent("connection-password", server.getPassword());
 
-            connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
-            connectorProperties.putIfAbsent("connection-url", jdbcUrl);
-            connectorProperties.putIfAbsent("allow-drop-table", "true");
+                queryRunner.installPlugin(new MySqlPlugin());
+                queryRunner.createCatalog("mysql", "mysql", connectorProperties);
 
-            queryRunner.installPlugin(new MySqlPlugin());
-            queryRunner.createCatalog("mysql", "mysql", connectorProperties);
+                copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
 
-            copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
-
-            return queryRunner;
+                return queryRunner;
+            }
+            catch (Throwable e) {
+                closeAllSuppress(e, queryRunner);
+                throw e;
+            }
         }
         catch (Throwable e) {
-            closeAllSuppress(e, queryRunner);
+            closeAllSuppress(e, server);
```

Comment:
since `server` is provided in arg, you shouldn't close it.

(we were doing things like this because we're calling this in ctor code, where we couldn't try-with-resources)